backend/test/core/usecase/AddTransactionToGroup.test.ts

```diff
@@ -73,71 +80,198 @@ describe("Add transaction to group (use case)", () => {
+        [secondUser.id],
+        -1
+      );
 
       // Assert
       expect(result).toStrictEqual({
         success: false,
         error: "Non positive transaction was submitted",
       });
     });
+
+    test("Given a group composed of one user, when adding a transaction from this user to this user, an error 'A GroupMember can not be exactly payer and the only recipient' should be returned", async () => {
+      // Arrange
+      const user = new GroupMember(1, "John");
+      const group = new Group(0, "group1", [user], []);
+      groupRepo.findById = vi.fn().mockResolvedValue(group);
+      const addTransactionToGroup = new AddTransactionToGroup(groupRepo);
+
+      // Act
+      const result = await addTransactionToGroup.execute(
+        group.id,
+        user.id,
+        [user.id],
+        1
+      );
+
+      // Assert
+      expect(result).toStrictEqual({
+        success: false,
+        error: "A GroupMember can not be exactly payer and the only recipient",
+      });
+    });
+
+    test("Given a group composed of one user, when adding a transaction from this user to another user, an error 'One recipient does not belong to the group or is duplicated' should be returned", async () => {
+      // Arrange
+      const user = new GroupMember(1, "John");
+      const otherUser = new GroupMember(2, "Dave");
+      const group = new Group(0, "group1", [user], []);
+      groupRepo.findById = vi.fn().mockResolvedValue(group);
+      const addTransactionToGroup = new AddTransactionToGroup(groupRepo);
+
+      // Act
+      const result = await addTransactionToGroup.execute(
+        group.id,
+        user.id,
+        [otherUser.id],
+        1
+      );
+
+      // Assert
+      expect(result).toStrictEqual({
+        success: false,
+        error: "One recipient does not belong to the group or is duplicated",
+      });
+    });
+
+    test("Given a group composed of two users, when adding a transaction from this user the other user two times, an error 'One recipient does not belong to the group or is duplicated' should be returned", async () => {
+      // Arrange
+      const user = new GroupMember(1, "John");
+      const otherUser = new GroupMember(2, "Dave");
+      const group = new Group(0, "group1", [user, otherUser], []);
+      groupRepo.findById = vi.fn().mockResolvedValue(group);
+      const addTransactionToGroup = new AddTransactionToGroup(groupRepo);
+
+      // Act
+      const result = await addTransactionToGroup.execute(
+        group.id,
+        user.id,
+        [otherUser.id, otherUser.id],
+        1
+      );
+
+      // Assert
+      expect(result).toStrictEqual({
+        success: false,
+        error: "One recipient does not belong to the group or is duplicated",
+      });
+    });
+
+    test("Given a group composed of two users, when adding a transaction from John to [John, Dave, Dave] , an error 'One recipient does not belong to the group or is duplicated' should be returned", async () => {
```

Comment:
This test is quite similar to the previous one, is it really necessary ?

Reply:
the previous one was removed